# Tâche 5

In [132]:
import numpy as np
from mp_api.client import MPRester
from pymatgen.core.operations import SymmOp
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.electronic_structure.plotter import BSPlotter
from jupyter_jsmol.pymatgen import quick_view
from plotly.subplots import make_subplots
import plotly.graph_objects as go

In [133]:
with MPRester("MVjrPtjUJ4K2IdZ4zJmL2QqGBqEUCt6C") as m:
    prim_struc = m.get_structure_by_material_id("mp-22660")
conv_struc = SpacegroupAnalyzer(prim_struc).get_conventional_standard_structure()
symmops = SpacegroupAnalyzer(conv_struc).get_space_group_operations()
sites = conv_struc.sites

Retrieving MaterialsDoc documents:   0%|          | 0/1 [00:00<?, ?it/s]

In [136]:
# Représentation 3D de InAg02
view = quick_view(conv_struc)
display(view)
# red : 02-
# grey : Ag+
# brown : In3-

JsmolView(layout=Layout(align_self='stretch', height='400px'))

#### Petit rappel: Les opérations de symétrie s'applique de la manière suivante : (matrice de rotation de l'opération de symétrie) * (coordonnées réduites de l'atome) + (vecteur de translation de l'opération de symétrie).

# Première symétrie

In [137]:
i = 1
print(symmops[i],"\n")

Rot:
[[-1.  0.  0.]
 [ 0. -1.  0.]
 [ 0.  0. -1.]]
tau
[0. 0. 0.] 



Comme premier élement de symétrie, nous avons une inversion (voir représentation jsmol) qui transforme les coordonnées (x,y,z) en (-x,-y,-z) avec la matrice de rotation : $\begin{pmatrix}
-1 & 0 & 0\\
0 & -1 & 0\\
0 & 0 & -1
\end{pmatrix}$ et le vecteur de translation : $\begin{pmatrix}
0 & 0 & 0
\end{pmatrix}$.

In [138]:
l = 4
at1 = sites[l]
op1 = symmops[i]

pos_init1 = at1.frac_coords
print("Position initiale:",pos_init1)

pos_final1 = op1.operate(pos_init1)
print("Position finale après symétrie:",pos_final1)

Position initiale: [0.66666667 0.33333333 0.33333333]
Position finale après symétrie: [-0.66666667 -0.33333333 -0.33333333]


Le centre d'inversion de l'opération de symétrie est un atome de Ag+ et l'opération d'inversion est appliquée à un atome In3- qui passe des coordonnées réduites $\begin{pmatrix}
0.66666667 & 0.33333333 & 0.33333333
\end{pmatrix}$
aux coordonnées réduites $\begin{pmatrix}
-0.66666667 & -0.33333333 & -0.33333333
\end{pmatrix}$.

In [139]:
# Représentation 3D par jsmol + image pour github
view1 = quick_view(conv_struc)
display(view1)
view1.script('draw SYMOP ' + str(i+1) + ' {atomno = ' + str(4) + '}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

![Symétrie d'inversion](Image/Inversion.png)

# Deuxième symétrie

In [140]:
j = 6
print(symmops[j],"\n")

Rot:
[[ 0.  1.  0.]
 [ 1.  0.  0.]
 [ 0.  0. -1.]]
tau
[0. 0. 0.] 



Comme deuxième élement de symétrie, nous avons une réflexion avec un plan miroir $\begin{pmatrix}
1 & \bar{1} & 0
\end{pmatrix}
$ qui transforme les coordonnées (x,y,z) en (y,x,-z) avec la matrice de rotation : $\begin{pmatrix}
0 & 1 & 0\\
1 & 0 & 0\\
0 & 0 & -1
\end{pmatrix}$ et le vecteur de translation : $\begin{pmatrix}
0 & 0 & 0
\end{pmatrix}$.

In [141]:
m = 6
at2 = sites[m]
op2 = symmops[j]

pos_init2 = at2.frac_coords
print("Position initiale:",pos_init2)
 
pos_final2 = op2.operate(pos_init2)
print("Position finale après symétrie:",pos_final2)

Position initiale: [0.66666667 0.33333333 0.22351079]
Position finale après symétrie: [ 0.33333333  0.66666667 -0.22351079]


L'opération de symétrie de plan miroir est appliquée à un atome Ag+ qui passe des coordonnées réduites $\begin{pmatrix}
0.66666667 & 0.33333333 & 0.22351079
\end{pmatrix}$
aux coordonnées réduites $\begin{pmatrix}
0.33333333 & 0.66666667 & -0.22351079
\end{pmatrix}$.

In [143]:
# Représentation 3D par jsmol + image pour github
view2 = quick_view(conv_struc)
display(view2)
view2.script('draw SYMOP ' + str(j + 1) + ' {atomno = ' + str(m + 1) + '}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

![Symétrie de réflexion](Image/Réflexion.png)

# Troisième symétrie

In [144]:
k = 24
print(symmops[k],"\n")

Rot:
[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]
tau
[0.33333333 0.66666667 0.66666667] 



Comme troisième élement de symétrie, nous avons une translation simple qui transforme les coordonnées (x,y,z) en (x+0.33333333,y+0.66666667,z+0.66666667) avec la matrice de rotation : $\begin{pmatrix}
1 & 0 & 0\\
0 & 1 & 0\\
0 & 0 & 1
\end{pmatrix}$ qui est la matrice unité et le vecteur de translation : $\begin{pmatrix}
0.33333333 & 0.66666667 & 0.66666667
\end{pmatrix}$.

In [145]:
n = 7
at3 = sites[n]
op3 = symmops[k]

pos_init3 = at3.frac_coords
print("Position initiale:",pos_init3)
 
pos_final3 = op3.operate(pos_init3)
print("Position finale après symétrie:",pos_final3)

Position initiale: [0.         0.         0.10982254]
Position finale après symétrie: [0.33333333 0.66666667 0.77648921]


L'opération de symétrie de translation simple est appliquée à un atome O2- qui passe des coordonnées réduites $\begin{pmatrix}
0. & 0. & 0.10982254
\end{pmatrix}$
aux coordonnées réduites $\begin{pmatrix}
0.33333333 & 0.66666667 & 0.77648921
\end{pmatrix}$.

In [146]:
# Représentation 3D par jsmol + image pour github
view3 = quick_view(conv_struc)
display(view3)
view3.script('draw SYMOP ' + str(k + 1) + ' {atomno = ' + str(n + 1) + '}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

![Symétrie de translation](Image/Translation.png)